drivers/i2c/i2c_litex.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2019 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include "i2c_bitbang.h"
+
+#define SCL_BIT_POS                0
+#define SDA_DIR_BIT_POS            1
+#define SDA_BIT_W_POS              2
+#define SDA_BIT_R_POS              0
+
+#define SDA_DIR_OUTPUT             1
+#define SDA_DIR_INPUT              0
+
+#define HIGH_STATE_ON_I2C_LINES    0x7
+
+struct i2c_litex_cfg {
+	volatile u32_t *w_reg;
+	volatile u32_t *r_reg;
+};
+
+#define GET_I2C_CFG(dev)						     \
+	((struct i2c_litex_cfg *) dev->config->config_info)
+
+#define GET_I2C_BITBANG(dev)						     \
+	((struct i2c_bitbang *) dev->driver_data)
+
+static inline void set_bit(volatile u32_t *reg, u32_t bit, u32_t val)
+{
+	u32_t mask = BIT(bit);
+
+	if (val) {
+		*reg |= mask;
+	} else {
+		*reg &= ~mask;
+	}
+}
+
+static inline int get_bit(volatile u32_t *reg, u32_t bit)
+{
+	u32_t mask = BIT(bit);
+
+	return !!((*reg) & mask);
+}
+
+void i2c_litex_bitbang_set_scl(void *context, int state)
```

Comment:
This should be `static`. Similarly all the other functions defined in this file. There's no need for their names to be seen outside.

Reply:
Changed as suggested for all functions in the file